### DISCUSSION TITLE
enhancement: Split Movement Speed and Attack Speed

### DISCUSSION THREAD

**Discussion by Uyarrr** [AscensionGameDev/Intersect-Engine#1547] [Ideas]
**Is your feature request related to a problem? Please describe.**
It's really weird and annoying that we have Speed stat for both Movement and Attack Speed. It's really bad, limiting hard and pointless.

**Describe the solution you'd like**
Instead of Speed, Should be Movement Speed and Attack Speed stats, or something called different I don't know. So we can have spells like grants attack speed, or like items that grants attack speed or movement speed only.

**Comment by WeylonSantana** [comment]
It's worth remembering that in this, the NPC entity uses movement speed as a speed attack, and it should be changed too, must be separated.

**Comment by SilverShardOnline** [comment]
I think this is a 100% needed feature and probably the most complained about issue aside from GUI editing.

**Comment by Arufonsu** [comment]
We don't really need to touch the speed stat too much, _probably rename it to "Movement Speed (MSPD)" or something similar for sure_, but as overall i think it's good as it is, i personally use a custom solution where the animation frames speed up as the character runs faster (and the other way around when slower), the speed stat works pretty well for this, and i _could_ share over a way to make this possible so the frames speed goes up and down dynamically and smoothly in relation with the MSPD stat. 

So, in the other hand what we need to add is the additional stat "Attack Speed" (ASPD) and replace MSPD with it around the weapon and attack speed "buff" methods. Then, we have to tweak things a little further, which means touching the editor (which nobody wants to touch anymore because we want to walk away from WinForms for good) and change some item/weapon configs around the dropdown options for attack speed:   static (ms)   ; ASPD%

![imagen](https://user-images.githubusercontent.com/17498701/226138000-76264fbd-9920-4967-8ac0-4dfb7eed0446.png)

![imagen](https://user-images.githubusercontent.com/17498701/226139940-0aad346a-b5b7-473e-bd31-d4bdf6ba569d.png)

We also need to change all these groups where stats and bonuses are set (concept):

![imagen](https://user-images.githubusercontent.com/17498701/226140149-45d27f2e-1d13-485f-b82e-0bc7f215d838.png)

**Comment by TalikanPME** [comment]
If it can help speed up the development of this system, I had it implemented in our game, and the sources are available here (v7.1 or 7.2) : 
https://github.com/TalikanPDMO/PDMO/tree/feature-movespeed-attackspeed

The running speed is managed in the server's config.json.
